internal/js/modules/k6/browser/common/frame_manager.go

```diff
@@ -540,7 +540,33 @@ func (m *FrameManager) requestStarted(req *Request) {
 		frame.pendingDocument = &DocumentInfo{documentID: req.documentID, request: req}
 		frame.pendingDocumentMu.Unlock()
 	}
-	m.logger.Debugf("FrameManager:requestStarted", "fmid:%d rurl:%s pdoc:nil", m.ID(), req.URL())
+
+	if !m.page.hasRoutes() {
+		return
+	}
+
+	route := NewRoute(m.logger, m.page.mainFrameSession.networkManager, req)
+	routeHandlers := m.page.getRoutes()
```

Comment:
I don't think this will work. We're returning the same reference to `p.routes` which could be amended by another thread.

Might need to bring the mutex into the caller i.e. in `FrameManager.requestStarted`. My concern there though is if the lock `routesMu` is being held in `requestStarted`, would that cause a deadlock? Maybe we need to something similar to in [urlTagName](https://github.com/grafana/k6/blob/80553b08ca52814a52ec4c970f464db622895d3d/internal/js/modules/k6/browser/common/page.go#L464-L482), where we unlock the lock before calling the handler.